src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
 package core.basesyntax;
 
+import static java.lang.Math.abs;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final int DEFAULT_INITIAL_CAPACITY = 16;
+    private final double DEFAULT_LOAD_FACTOR = 0.75;
+    private HashObject<K, V>[] hashMap;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size = 0;
+
+    public MyHashMap() {
+        hashMap = new HashObject[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    public class HashObject<K, V> {
+        private K key;
+        private V value;
+        private HashObject<K, V> next;
+
+        public HashObject(K key, V value, HashObject<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public void enlarge() {
+        K[] listOfKeys = (K[]) new Object[size];
+        V[] listOfValues = (V[]) new Object[size];
+        for (int i = 0; i < size; i++) {
+            listOfKeys[i] = keySet()[i];
+            listOfValues[i] = getValue(keySet()[i]);
+        }
+        hashMap = new HashObject[capacity * 2];
+        size = 0;
+        capacity = capacity * 2;
+
+        for (int i = 0; i < listOfKeys.length; i++) {
+            put(listOfKeys[i], listOfValues[i]);
+        }
+    }
+
+    public int keyIndex(K key) {
+        if (key == null) {
+            return 0;
+        } else {
+            return abs(key.hashCode()) % capacity;
+        }
+    }
+
+    public K[] keySet() {
```

Comment:
The keySet() method could be implemented more efficiently. Currently, it creates a new array each time it is called which can be costly in terms of performance. Consider storing the keys in a separate field and updating it each time a put() operation is performed.